practical-penguins/trivia_tavern/trivia_runner/views.py

```diff
@@ -0,0 +1,93 @@
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.http import HttpResponseNotFound
+from django.shortcuts import render, get_object_or_404
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import DeleteView, ListView
+from django.contrib import messages
+
+from trivia_builder.models import TriviaQuestion
+from trivia_runner.models import ActiveTriviaQuiz
+from twilio_messenger.views import SMSBot
+
+from .forms import PhoneNumberForm
+
+
+class ActiveTriviaQuizListView(ListView):
+    model = ActiveTriviaQuiz
+    template_name = 'active_sessions.html'
+    context_object_name = 'active_quizzes'
+    ordering = ['-start_time']
+    paginate_by = 10
+
+
+def setup(request, active_trivia_quiz):
+    print(request.POST)
+    if 'phone_number' in request.POST:
+        form = PhoneNumberForm(request.POST)
+        if form.is_valid():
+            form.save()
+            number = form.cleaned_data.get('phone_number').__str__()
+            messages.success(request, f'Invite sent to {number}!')
+            SMSBot.send_quiz_invite(number, active_trivia_quiz)
+    else:
+        form = PhoneNumberForm()
+
+    return render(request, 'activequiz_setup.html', {'active_trivia_quiz': active_trivia_quiz, 'form': form})
+
+
+def times_up(request, active_trivia_quiz):
+    SMSBot.player_timeout(active_trivia_quiz)
+    cur_question = TriviaQuestion.objects.get(quiz=active_trivia_quiz.trivia_quiz,
+                                              question_index=active_trivia_quiz.current_question_index)
+    return render(request, 'activequiz_question.html',
+                  {'active_trivia_quiz': active_trivia_quiz, 'cur_question': cur_question})
+
+
+def question(request, active_trivia_quiz):
+    SMSBot.send_all_questions(active_trivia_quiz)
+    cur_question = TriviaQuestion.objects.get(quiz=active_trivia_quiz.trivia_quiz,
+                                              question_index=active_trivia_quiz.current_question_index)
+    return render(request, 'activequiz_question.html',
+                  {'active_trivia_quiz': active_trivia_quiz, 'cur_question': cur_question})
+
+
+def end_screen(request, active_trivia_quiz):
+    tally_results = SMSBot.calculate_results(active_trivia_quiz)
+    return render(request, 'activequiz_end.html',
+                  {'active_trivia_quiz': active_trivia_quiz, 'tally_results': tally_results})
+
+
+@csrf_exempt
+def active_trivia(request, pk):
```

Comment:
This definitely could use a docstring. But overall, nice work again on writing easy to follow code.

My only minor comment is that you might want to use specific fields of key/value pairs (via something like JSON payload) for anything more complex than this (e.g. if it wasn't just one of three potential options). Searching for specific words in the string works decently well for this case though.